deeplearning4j-core/src/main/java/org/deeplearning4j/nn/conf/preprocessor/output/NetworkOutputProcessor.java

```diff
@@ -0,0 +1,31 @@
+package org.deeplearning4j.nn.conf.preprocessor.output;
+
+import java.io.Serializable;
+
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/** NetworkOutputProcessor processes both final network output
+ * (during forward pass) and the labels (during backprop).<br>
+ * Can be used to do things like reshape output and labels, for
+ * example.
+ * @author Alex Black
+ */
+@JsonTypeInfo(use= JsonTypeInfo.Id.NAME, include= JsonTypeInfo.As.WRAPPER_OBJECT)
+@JsonSubTypes(value={
+        @JsonSubTypes.Type(value = RnnOutputProcessor.class, name = "rnnOutputProcessor"),
```

Comment:
To be consistent with other aspects of the serialized output, please change  `name="rnnOutputProcessor"` to `name="rnn"`.  The term "outputProcessor" is redundant.